src/OrchardCore.Modules/OrchardCore.Liquid/ScriptsMiddleware.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Linq;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using OrchardCore.DisplayManagement.Liquid;
+using OrchardCore.Entities;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Settings;
+using Fluid;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace OrchardCore.Liquid
+{
+    public class ScriptsMiddleware
```

Comment:
Why not use a static js file for now?
If you use IMemoryCache it would be to add a sliding expiration, or instead use a local variable. But sliding expiration is better. Also the response should have public cache headers so the client doesn't request it again.